game-core/src/main/java/games/strategy/engine/framework/map/download/ContentReader.java

```diff
@@ -85,24 +64,8 @@ void downloadToFile(final String uri, final File file) throws IOException {
     checkNotNull(file);
 
     try (FileOutputStream os = new FileOutputStream(file)) {
-      downloadInternal(
+      downloadAndApplyAction(
           uri, is -> os.getChannel().transferFrom(Channels.newChannel(is), 0L, Long.MAX_VALUE));
```

Comment:
Also just noticed this method exists, which simplifies this even further:
```suggestion
          uri, is -> Files.copy(is, file.toPath()));
```

Reply:
This seems safe, but I'm a bit nervous about committing updates without testing them, also nervous about increasing the test scope of the update. There is something to be said about reducing how many things that can go wrong.

Reply:
Updated: 46e80d077b40799943dbb809d82aadd3b9d23f95

But then did some quick testing, and hit:
```
Testing:
SEVERE: Failed to download: https://github.com/triplea-maps/diplomacy-map_skin1/archive/master.zip
java.nio.file.FileAlreadyExistsException: /tmp/triplea1756697996669965286tmp
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:94)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:215)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:219)
	at java.base/java.nio.file.Files.copy(Files.java:3066)
	at games.strategy.engine.framework.map.download.ContentReader.lambda$downloadToFile$0(ContentReader.java:65)
	at games.strategy.engine.framework.map.download.ContentReader.downloadAndApplyAction(ContentReader.java:50)
	at games.strategy.engine.framework.map.download.ContentReader.downloadToFile(ContentReader.java:65)
	at games.strategy.engine.framework.map.download.DownloadFile.lambda$newDownloadThread$2(DownloadFile.java:64)
	at java.base/java.lang.Thread.run(Thread.java:834)
```
And reverted (46e80d): 
c1dd7b9bcf7f524ffa1981d4558bb995cd26c90d

Reply:
Ah, in this case you need to pass https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/StandardCopyOption.html#REPLACE_EXISTING as third param, didn't knew that either

Reply:
It's a bit curious we're using the same temp file to begin with, it might have issues with concurrent file downloads. Let's put a pin in this one and update post-merge (for the sake of testing and focus)